tests/ui/async-await/async-fn/simple.rs

```diff
@@ -12,5 +15,7 @@ async fn call_asyncly(f: impl AsyncFn(i32) -> i32) -> i32 {
 }
 
 fn main() {
-    let fut = call_asyncly(|x| async move { x + 1 });
+    block_on::block_on(async {
+        call_asyncly(|x| async move { x + 1 }).await;
```

Comment:
Turns out that this wasn't caught because we weren't actually codegenning `<{closure} as AsyncFn>::async_call` here, since we weren't awaiting the future.

Makes me wish we had a first-class `async main` to make testing here easier.